### PR TITLE
Add CloudBluePSA documents

### PR DESCRIPTION
**Pull request by mohsad06**
This is a pull request for the documents required for the CloudBluePSA connector.

### CODE REVIEW

**Comment by Court72** [comment]
#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Comment by JasonWHowell** [comment]
@mohsad06 please pull from https://github.com/MicrosoftDocs/powerquery-docs/ main to get the latest files, and help solve the conflict. I believe the TOC.yml file was renamed to toc.yml (lowercase), hence the conflict.

**Comment by mohsad06** [comment]
I renamed the table of contents file's name to toc.yml. Please let me know if anything else is needed.

**Comment by mohsad06** [comment]
I resolved the conflicts now. Please let me know if anything else is needed.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: Power Query CloudBlue PSA connector
+description: Provides basic information and prerequisites for the operation of connector in conjuction with the API.
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/cloudbluepsa.md:3]: nit: this should include CloudBlue PSA

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+author: bezhan-msft
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/cloudbluepsa.md:4]: This should be your (CloudBlue) GitHub author.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+ms.topic: conceptual
+ms.date: 07/03/2022
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/cloudbluepsa.md:8]: Date should be updated to 2023.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ms.author: bezhan 
+LocalizationGroup: reference
+---
+
+# CloudBluePSA
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/cloudbluepsa.md:13]: This needs to be "CloudBluePSA (Beta)"

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+>[!Note]
+The following connector article is provided by CloudBluePSA, the owner of this connector and a member of the Microsoft Power Query Connector Certification Program. If you have questions regarding the content of this article or have changes you would like to see made to this article, visit the CloudBluePSA website and use the support channels there.
+
+
+## Summary
+
+| Item | Description |
+| ---- | ----------- |
+| Release State | General Availability |
+| Products | Power BI (datasets)
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/cloudbluepsa.md:24]: This needs to include Power BI (Dataflows) as well.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+| Authentication Types Supported | API Key |
+| Function Reference Documentation | [CloudBluePSA Connector.Help](https://help.harmonypsa.com/articles/#!cloudblue-psa-4-28-publication/using-the-power-bi-connector)
+| | |
+## Prerequisites
+Before you can use the CloudBluePSA connector, you must have an CloudBlue PSA instance (username/password) and an API key.
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/cloudbluepsa.md:29]: nit: link to where to get this instance and key

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+7. Expand the **Column1.1** column. This time, on the list of columns all the grid columns included in the filter are displayed. Select as many columns as required then select **OK**. 
+All the selected data is now displayed, and can be reshaped and used to create reports as required.
+
+    ![Expand Data Column1.1.](./media/cloudbluepsa/expand1.1.png)
+
+
+8. Select **Close & Apply**. You can now start using your data.
+
+
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/cloudbluepsa.md:75]: This needs a Connect to CloudBlue PSA in Power Query Online section.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+7. Expand the **Column1.1** column. This time, on the list of columns all the grid columns included in the filter are displayed. Select as many columns as required then select **OK**. 
+All the selected data is now displayed, and can be reshaped and used to create reports as required.
+
+    ![Expand Data Column1.1.](./media/cloudbluepsa/expand1.1.png)
+
+
+8. Select **Close & Apply**. You can now start using your data.
+
+ 
+  ## Next Steps
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/cloudbluepsa.md:76]: Next Steps is autogenerated. This should be "additional resources"

File: powerquery-docs/connectors/index.md [hunk 9]
```diff
@@ -73,6 +73,7 @@ The connectors are listed in alphabetical order in separate tables for each lett
...
 | ![Cherwell (Beta).](media/index/cherwell-64.png)<br/>**Cherwell (Beta)**<br/>By Cherwell                            | ![Not supported - Cherwell (Beta) in Excel](media/index/no.png)  | ![Supported - Cherwell (Beta) in Power BI (Datasets)](media/index/yes.png) | ![Supported - Cherwell (Beta) in Power BI (Dataflows)](media/index/yes.png)  | ![Not supported - Cherwell (Beta) in Power Apps (Dataflows)](media/index/no.png)  | ![Not supported - Cherwell (Beta) in Customer Insights (Dataflows)](media/index/no.png)  | ![Not supported - Cherwell (Beta) in Analysis Services](media/index/no.png)  |
+| [![CloudBluePSA.](media/index/cloudbluepsa-64.png)<br/>**CloudBluePSA**<br/>](./cloudbluepsa.md)By CloudBlue PSA | ![Not supported - CloudBluePSA in Excel](media/index/no.png) | ![Supported - CloudBluePSA in Power BI (Datasets)](media/index/yes.png) | ![Not supported - CloudBluePSA in Power BI (Dataflows)](media/index/no.png) | ![Not supported - CloudBluePSA in Power Apps (Dataflows)](media/index/no.png) | ![Not supported - CloudBluePSA in Customer Insights (Dataflows)](media/index/no.png) | ![Not supported - CloudBluePSA in Analysis Services](media/index/no.png) |
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/index.md:76]: this needs to include (Beta) in the name.

File: powerquery-docs/toc.yml [hunk 10]
```diff
@@ -360,6 +360,8 @@
         href: connectors/bqecore.md
       - name: "CData Connect Cloud (Beta)"
         href: connectors/cdata-connect-cloud.md
+      - name: "CloudBluePSA"
```
Comment (bezhan-msft) [inline powerquery-docs/toc.yml:363]: needs to include Beta

**Review by bezhan-msft — CHANGES_REQUESTED** [review]
@DougKlopfenstein to also review for style.

File: powerquery-docs/connectors/cloudbluepsa.md (hunk 1 above)
Comment (LefterisIoan) [reply powerquery-docs/connectors/cloudbluepsa.md:3]: amended

File: powerquery-docs/connectors/cloudbluepsa.md (hunk 3 above)
Comment (LefterisIoan) [reply powerquery-docs/connectors/cloudbluepsa.md:8]: amended

File: powerquery-docs/connectors/cloudbluepsa.md (hunk 2 above)
Comment (LefterisIoan) [reply powerquery-docs/connectors/cloudbluepsa.md:4]: amended

File: powerquery-docs/connectors/cloudbluepsa.md (hunk 4 above)
Comment (LefterisIoan) [reply powerquery-docs/connectors/cloudbluepsa.md:13]: amended

File: powerquery-docs/connectors/cloudbluepsa.md (hunk 7 above)
Comment (LefterisIoan) [reply powerquery-docs/connectors/cloudbluepsa.md:75]: Hi Bob, I am not sure what to add here. Could you please provide an example? Thank you

**Comment by mohsad06** [comment]
Requested changes are committed and pushed. Please let us know if anything further is needed.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
+---
+title: Power Query CloudBlue PSA connector
```
Comment (DougKlopfenstein) [inline powerquery-docs/connectors/cloudbluepsa.md:2]: In some cases, the connector is referred to as "CloudBlue PSA" (with a space), and other times as "CloudBluePSA" (without a space). This name needs to be consistent throughout the article.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+## Connect to CloudBlue PSA
+
+To connect to CloudBlue PSA data:
+
+1. Select **Get Data** from the **Home** ribbon in Power BI Desktop. 
+
+2. Select the **Search** box and start typing **CloudBluePSA**. Then select **CloudBluePSA** from the list on the right, and select **Connect**.
+
+    ![Screenshot of the Get Data dialog for CloudBlue PSA.](./media/cloudbluepsa/getdata.png)
```
Comment (DougKlopfenstein) [inline powerquery-docs/connectors/cloudbluepsa.md:40]: Image names need to be all lowercase, with a dash between words. So change the name of this image to get-data.png (both here and in the file name).

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+
+3. In the **Retrieve all pages of data** window that appears, copy and paste the **URL** generated on the **GET** side of the API endpoint of your choice. Then in **Filter**, copy and paste the constructed filter on the same API endpoint. For example:
+
+    ### URL: ### 
+    https://INSTANCE_URL/webapi/v1.3/tickets/getticketsreport
+    
+    ### Filter ### 
+    {  "pageno": 1,  "pagesize": 10,  "filters": [{"field": "Type", "op": "cn", "data": "n"}], "filterop": "AND",  "sortcolumn": "Name",  "sortorder": "Asc",  "resulttype": "Json",  "includeccysymbols": true,  "gridcolumns": "RefNumber,Type,Name"}
+
+    ![Screenshot showing Retrieve all pages of data for CloudBlue PSA.](./media/cloudbluepsa/retrieveallpagesofdata.png)
```
Comment (DougKlopfenstein) [inline powerquery-docs/connectors/cloudbluepsa.md:50]: Change the name of this image to "retrieve-all-pages-of-data.png" (both here and in the file name).

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+5. You need to select **Transform Data** and this selection opens the Power Query editor.  
+
+6. Expand the **Data** column, then select **OK**.
+
+    ![Screenshot of the Expand Data Column dialog.](./media/cloudbluepsa/expand1.png)
+
+7. Expand the **Column1.1** column. This time, on the list of columns all the grid columns included in the filter are displayed. Select as many columns as required, and then select **OK**. 
+All the selected data is now displayed, and can be reshaped and used to create reports as required.
+
+    ![Screenshot showing the expanded Data Column1.1.](./media/cloudbluepsa/expand1.1.png)
```
Comment (DougKlopfenstein) [inline powerquery-docs/connectors/cloudbluepsa.md:65]: Change the name of this image to "expand-1-1.png (both here and in the file name).

File: powerquery-docs/connectors/cloudbluepsa.md (hunk 7 above)
Comment (DougKlopfenstein) [reply powerquery-docs/connectors/cloudbluepsa.md:68]: You can use the [Azure Blob Storage](https://learn.microsoft.com/power-query/connectors/azure-blob-storage) "Connect to Azure Blob Storage from Power Query Online" section as an example.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+8. Select **Close & Apply**. You can now start using your data.
+ 
+  ## Additional Resources
+ You might also find the following CloudBlue PSA information useful:
+
+ * [CloudBlue PSA Help Centre](https://help.harmonypsa.com/home/en-gb/)
+ * [Connecting PowerBI to the API](https://help.harmonypsa.com/articles/#!cloudblue-psa-4-28-publication/connecting-powerbi-to-the-api)
+ * [API](https://help.harmonypsa.com/articles/#!cloudblue-psa-4-28-publication/api)
+    * [Creating an API Key](https://help.harmonypsa.com/articles/#!cloudblue-psa-4-28-publication/creating-an-api-key)
+    * [Using the API](https://help.harmonypsa.com/articles/#!cloudblue-psa-4-28-publication/using-the-api)
```
Comment (DougKlopfenstein) [inline powerquery-docs/connectors/cloudbluepsa.md:76]: There are two unused images in this pull request: feedkey.png and transform.png. If these images aren't going to be used in the article, remove them.

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+# CloudBluePSA (Beta)
+
+>[!Note]
+The following connector article is provided by CloudBluePSA, the owner of this connector and a member of the Microsoft Power Query Connector Certification Program. If you have questions regarding the content of this article or have changes you would like to see made to this article, visit the CloudBluePSA website and use the support channels there.
+
+## Summary
+
+| Item | Description |
+| ---- | ----------- |
+| Release State | General Availability |
```
Comment (DougKlopfenstein) [inline powerquery-docs/connectors/cloudbluepsa.md:19]: This connector appears to be a "Beta", so change "General Availability" to "Beta".

**Comment by mohsad06** [comment]
@DougKlopfenstein Dear Doug, I have applied the changes you requested. Please let me know if anything further is required.

File: powerquery-docs/connectors/cloudbluepsa.md (hunk 7 above)
Comment (DougKlopfenstein) [reply powerquery-docs/connectors/cloudbluepsa.md:68]: Did you follow the instructions in the Power Query documentation article called [Where to get data](https://learn.microsoft.com/power-query/where-to-get-data), specifically the section titled [Get data in Power BI service](https://learn.microsoft.com/power-query/where-to-get-data#get-data-in-power-bi-service)? If not, start there, and you should be able to run Power Query Online.

If you can't access the Power BI service, just use the [free trial version](https://powerbi.microsoft.com/getting-started-with-power-bi/). Go to that link, then select "Free trial" in the upper right hand corner.

Comment (mohsad06) [reply powerquery-docs/connectors/cloudbluepsa.md:95]: The changes for this are now pushed.

**Review by DougKlopfenstein — APPROVED** [review]

**Review by bezhan-msft — APPROVED** [review]

**Comment by bezhan-msft** [comment]
#sign-off

File: powerquery-docs/connectors/cloudbluepsa.md [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+author: mohsad06
+ms.topic: conceptual
+ms.date: 3/7/2023
+ms.author: bezhan 
+---
+
+# CloudBluePSA (Beta)
+
+>[!Note]
+The following connector article is provided by CloudBluePSA, the owner of this connector and a member of the Microsoft Power Query Connector Certification Program. If you have questions regarding the content of this article or have changes you would like to see made to this article, visit the CloudBluePSA website and use the support channels there.
```
Comment (Court72) [inline powerquery-docs/connectors/cloudbluepsa.md:13]:
```suggestion
>The following connector article is provided by CloudBluePSA, the owner of this connector and a member of the Microsoft Power Query Connector Certification Program. If you have questions regarding the content of this article or have changes you would like to see made to this article, visit the CloudBluePSA website and use the support channels there.
```

**Comment by Court72** [comment]
#hold-off

**Comment by mohsad06** [comment]
Dear @Court72 I've applied the changes you requested.

**Comment by Jak-MS** [comment]
@mohsad06 - thanks for making those changes. We're going to move the commits to the private repo so we can use tools there (like staging and Acrolinx). Will post here when this is ready to merge.

fyi @bezhan-msft

**Pull request merged**